src/lib/cert/x509/x509path.cpp

```diff
@@ -143,6 +141,62 @@ check_chain(const std::vector<X509_Certificate>& cert_path,
          if(!trusted_hashes.count(subject.hash_used_for_signature()))
             status.insert(Certificate_Status_Code::UNTRUSTED_HASH);
          }
+
+      const NameConstraints& name_constr = issuer.name_constraints();
+
+      if(!name_constr.permitted().empty() || !name_constr.excluded().empty())
+         {
+         if(!issuer.is_CA_cert() || !issuer.is_critical("X509v3.NameConstraints"))
+            cert_status.at(i).insert(Certificate_Status_Code::NAME_CONSTRAINT_ERROR);
+
+         // Check that all subordinate certs pass the name constraint
+         for(int j = i; j >= 0; j--)
```

Comment:
I see now, size_t doesn't work and ithe cert chain size will not be at risk of overflowing signed int so this is harmless in practice. But it causes compiler warnings due to the signed/unsigned compare. It does not look like the order that the constraints are checked matters, so why not `for(size_t j = 0; j <= i; ++j)`?